clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/AggKafkaConsumerService.java

```diff
@@ -100,6 +134,106 @@ public void stopInner() throws Exception {
+
+      /**
+       * Collect a list of SITs, whose version topic doesn't exist by checking {@link StoreIngestionTask#isProducingVersionTopicHealthy()},
+       * and this function will continue to check the version topic healthiness for a period of {@link nonExistingTopicIngestionTaskKillThresholdMs}
+       * to tolerate transient topic discovery issue.
+       */
+      Map<String, StoreIngestionTask> versionTopicIngestionTaskMappingForNonExistingTopic = new HashMap<>();
+      versionTopicStoreIngestionTaskMapping.forEach((vt, sit) -> {
+        try {
+          if (!sit.isProducingVersionTopicHealthy()) {
+            versionTopicIngestionTaskMappingForNonExistingTopic.put(vt, sit);
+            LOGGER.warn("The producing version topic:{} is not healthy", vt);
+          }
+        } catch (Exception e) {
+          LOGGER.error("Got exception while checking topic existence for version topic: {}", vt, e);
+        }
+      });
+      int maxAttempts =
+          (int) Math.ceil((double) nonExistingTopicIngestionTaskKillThresholdMs / nonExistingTopicRetryIntervalMs);
+      for (int cnt = 0; cnt < maxAttempts; ++cnt) {
+        Iterator<Map.Entry<String, StoreIngestionTask>> iterator =
+            versionTopicIngestionTaskMappingForNonExistingTopic.entrySet().iterator();
+        while (iterator.hasNext()) {
+          Map.Entry<String, StoreIngestionTask> entry = iterator.next();
+          String versionTopic = entry.getKey();
+          StoreIngestionTask sit = entry.getValue();
+          try {
+            if (sit.isProducingVersionTopicHealthy()) {
+              /**
+               * If the version topic becomes available after retries, remove it from the tracking map.
+               */
+              iterator.remove();
+              LOGGER.info("The producing version topic:{} becomes healthy", versionTopic);
+            }
+          } catch (Exception e) {
+            LOGGER.error("Got exception while checking topic existence for version topic: {}", versionTopic, e);
+          } finally {
+            Utils.sleep(nonExistingTopicRetryIntervalMs);
+          }
+        }
+      }
+
+      AtomicBoolean repairSomeIngestionTask = new AtomicBoolean(false);
+      versionTopicIngestionTaskMappingForNonExistingTopic.forEach((vt, sit) -> {
+        LOGGER.warn(
+            "The ingestion topics (version topic) are not healthy for "
+                + "store version: {}, will kill the ingestion task to try to unblock shared consumer",
+            vt);
+        /**
+         * The following function call will interrupt all the stuck {@link org.apache.kafka.clients.producer.KafkaProducer#send} call
+         * to non-existing topics.
+         */
+        sit.closeVeniceWriters(false);
+        killIngestionTaskRunnable.accept(vt);
+        repairSomeIngestionTask.set(true);
+        stuckConsumerRepairStats.recordIngestionTaskRepair();
+      });
+      if (!repairSomeIngestionTask.get()) {
+        LOGGER.error(
```

Comment:
curious, what this check for ? It means we found some stuck consumer, but no unhealthy topic ? should we init `repairSomeIngestionTask` to false only if `versionTopicIngestionTaskMappingForNonExistingTopic` size is > 0 ?
also nit: `repairSomeIngestionTask` rename to isRepairedSomeIngestionTask to indicate its a boolean.

Reply:
```
It means we found some stuck consumer, but no unhealthy topic
```
We want to track this case via metric and logging.

Regarding other minor comments, I can pick it up in another PR since perf diff is minor.